### PR TITLE
Feature/gdal datasources

### PR DESCRIPTION
**Pull request by milos-colic**

### CODE REVIEW

File: src/test/scala/com/databricks/labs/mosaic/datasource/GDALFileFormatTest.scala [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                "PDF",
+                "PNG",
+                "VRT",
+                "XMP",
+                "COG",
+                "GRIB",
+                "Zarr"
+            )
+        ) {
+            GDALFileFormat.getFileExtension(driver) should not be ""
```
Comment (edurdevic) [inline src/test/scala/com/databricks/labs/mosaic/datasource/GDALFileFormatTest.scala:143]: Maybe better test that should not be "UNSUPPORTED"

**Review by edurdevic — APPROVED** [review]
This is awesome @milos-colic !

File: src/test/scala/com/databricks/labs/mosaic/datasource/GDALFileFormatTest.scala (hunk 1 above)
Comment (milos-colic) [reply src/test/scala/com/databricks/labs/mosaic/datasource/GDALFileFormatTest.scala:143]: fair point, will update

File: src/main/scala/com/databricks/labs/mosaic/datasource/Utils.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+     *   sequence of values.
+     * @return
+     *   Spark SQL row.
+     */
+    def createRow(values: Seq[Any]): InternalRow = {
+        InternalRow.fromSeq(
+            values.map {
+                case null           => null
+                case b: Array[Byte] => b
+                case v: Array[_]    => new GenericArrayData(v)
```
[the remark this answers is not in the document]
Comment (milos-colic) [reply src/main/scala/com/databricks/labs/mosaic/datasource/Utils.scala:24]: will add a test line

**Pull request merged**